crypto/http/http_lib.c

```diff
@@ -48,7 +49,7 @@ int OSSL_HTTP_parse_url(const char *url, char **phost, char **pport,
     if ((buf = OPENSSL_strdup(url)) == NULL)
         goto err;
 
-    /* Check for initial colon */
+    /* check for optional http[s]:// prefix */
     p = strchr(buf, ':');
     if (p == NULL || (size_t)(p - buf) > https_len) {
```

Comment:
It this supposed to be allowed?

anynamebiggerthan5chars://something.com

Is this related to @ ??

Reply:
For HTTP(S) URLs only "http://" or "https://" may be used.
This PR does not change this behavior but makes it explicit in the documentation.

Reply:
There is nothing **in the existing code** to stop you putting in a bogus URL as I have done - and it will just skip it - so the question is should it do this a bit better?

Reply:
Good point.
I've meanwhile made sure that only the schemes "http" and "https" are allowed.
Doing so I was able to actually simplify the code.
Moreover, I've added various tests for `OSSL_HTTP_parse_url()`.